crates/nu-cli/src/cli.rs

```diff
@@ -568,6 +568,13 @@ pub async fn cli(
 
         rl.set_edit_mode(edit_mode);
 
+        let max_history_size = config::config(Tag::unknown())?
```

Comment:
As long as you tested that setting it inside of the loop works okay, I'm fine with landing it.

Reply:
Yes, I have tested it locally in release and in debug mode. Please see the following screenshot:
![2020-05-18-1589833735_screenshot_2560x1440](https://user-images.githubusercontent.com/14004106/82256655-f9a52700-9913-11ea-807c-0b2362993698.jpg)

_Note: It appears rustyline capping history to `history_size + 1`. On 0.14.0 I am seeing `history | count` return `101 consistently even though the [default is 100](https://github.com/kkawakam/rustyline/blob/be20cfac991e7bc30d59b712b219438f5222f6e5/src/config.rs#L151). I believe that should be addressed upstream._

Reply:
(Disregard that note, I see now the extra line is coming from a history versioning scheme they are using: https://github.com/lightclient/rustyline/blob/abffcbb552b1973b399e838b400f280b65192c74/src/history.rs#L141)